websock/types.nim

```diff
@@ -112,6 +115,26 @@ type
     factory*: ExtFactoryProc
     clientOffer*: string
 
+  # client exec order:
+  #   1. append to request header
+  #   2. verify response header
+  # server exec order:
+  #   1. verify request header
+  #   2. append to response header
+  # ------------------------------
+  # Handshake exec order:
+  # 1. client append to request header
+  # 2. server verify request header
+  # 3. server reply with response header
+  # 4. client verify response header from server
+  HdrPlugin* = ref object of RootObj
```

Comment:
I'd say that `plugin` is a bit misleading as it conflicts with native websocket plugins nomenclature - how about `hook`?

Reply:
no problem. `hook` is more ergonomic and easier to type.

But I have other problem during the second step:
```text
1. client append to request header
2. server verify request header                    # <---- this second step
3. server reply with response header
4. client verify response header from server
```

There are cases where authentication not replied with `Http101 + response header` but `Http401` or other Http Error code. Still thinking the best approach to this situation.

Reply:
> There are cases where authentication not replied with Http101 + response header but Http401 or other Http Error code. Still thinking the best approach to this situation.

What specifically is making handing this case difficult? Is it propagating the error to the caller or something else I'm missing?